src/activity/exec.rs

```diff
@@ -0,0 +1,113 @@
+use std::path::{PathBuf};
+use std::net::{SocketAddr, IpAddr, TcpStream, UdpSocket};
+use std::process::{Command, Stdio};
+use std::fs::{self, File, OpenOptions};
+use std::io::{Result, Error, ErrorKind, Write};
+use std::iter;
+use chrono::{Utc};
+
+use super::ActivityContext;
+use super::log::{ActivityLog, ActionLogInfo, ProcessLogInfo, FileLogInfo, NetLogInfo};
+use super::spec::{FileAction, FileOp, FileType, IpProto};
+
+pub fn process_action(name: &str, args: &Vec<String>, ctx: &ActivityContext) -> Result<ActivityLog> {
+    let proc = Command::new(name)
+        .args(args)
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .spawn()?;
+
+    let mut cmdvec = vec![name.to_string()];
+    cmdvec.extend(args.iter().cloned());
+
+    let mut ctxvec = vec![ctx.cmd.to_string()];
+    ctxvec.extend(ctx.args.iter().cloned());
```

Comment:
Instead of calling `args.iter().cloned()`, you can actually just call `args.clone()` since Vec<...> implements the `IntoIterator` trait.

Reply:
I think I pulled that construction out of Iterator's extend method documentation and didn't think too much about it, but just cloning would be simpler.